### PR TITLE
Show a more prominent warning in the documentation of components

### PR DESCRIPTION
**Pull request by javiereguiluz**
Fixes the problem explained in #16469.

### CODE REVIEW

**Comment by OskarStark** [comment]
Ping @epitre

**Review by 94noni — APPROVED** [review]
Nice addition imho
Sometime i think the same when i arrived on the not wanted doc (from search engine or google etc)

just a note, is’t the caution bloc a « too high »
Why not a note for example (just challenging as this notice is mainly for new user/beginer/new comer)

**Comment by javiereguiluz** [comment]
@94noni I thought about adding something lighter than a caution box ... but people don't read that.

We already have today some `seealso` boxes in some components; many people just ignore those. That's why I think it's better to not be subtle about this.

**Comment by 94noni** [comment]
rethinking of this, could it be great to move this PR new block (caution/notice/tip/whatever) after the catch phrase of the component ?

because, imho, reading the component name as title, then the block, then the catch phrase looks weird

**Comment by epitre** [comment]
Thanks for the ping @OskarStark  
I think it's great what @javiereguiluz did.  

Thank you very much!  
For the "not subtle" thing, I would use "caution" in the component documentation because we are in a symfony website, and one might expect a symfony framework related documentation.  
On the other side, in the symfony framework documentation, I would use "seealso" because some detailed information about the component might be found in the component focused documentation.  

My 2 cents.

**Comment by OskarStark** [comment]
@javiereguiluz if this is still a thing, please rebase on `6.3`, thanks 🌻

**Review by alexandre-daubois — APPROVED** [review]

**Comment by wouterj** [comment]
Sharing what I've said about this in our internal Slack channel:

I'm not sure about it for 2 reasons:
1. For most component docs that exists as of yet, we actively link to it from the framework docs to read more about features. I agree 100% to move this from component docs to framework docs and then warn people when visiting the component docs (or actually move them to a separate location, e.g. other part of the website with a different layout or the github README). But doing this now will I think introduce confusion, as features are simply only documented in the component docs. I'm afraid "you don't have to do most of the things explained here" is too vague to really guide people.
2. More importantly, I don't think the current seealso directive doesn't work because it isn't red and screaming, but simply because people don't see the top of the page. For one, we link to sections within the page (and then someone starts reading the whole article from there). On top of that, I'm pretty convinced most people skip everything except from code blocks and then start reading the paragraph before/after. So I'm afraid most people don't even see what is placed in the first fold of the page.

File: components/expression_language.rst [hunk 1]
```diff
@@ -1,9 +1,17 @@
...
 
-    The ExpressionLanguage component provides an engine that can compile and
-    evaluate expressions. An expression is a one-liner that returns a value
-    (mostly, but not limited to, Booleans).
+.. caution::
+
+    This article explains how to define and evaluate expressions in any PHP application
+    using the ExpressionLanguage component. If you are working on a Symfony application,
+    you don't have to do most of the things explained here.
+    Instead, read the expression syntax reference in :doc:`this other article <//reference/formats/expression_language>`,
```
Comment (alexandre-daubois) [inline components/expression_language.rst:9]:
```suggestion
    Instead, read the expression syntax reference in :doc:`this other article </reference/formats/expression_language>`,
```

**Review by alexandre-daubois — COMMENTED** [review]
About @wouterj's comment, maybe a reword could help? Something like "If you are working with the whole Symfony framework, this component is fully integrated with your application as explained in [page]" in a `.. note` section? Just an idea :slightly_smiling_face: 

We already have this in [this page](https://symfony.com/doc/current/components/messenger.html) and it really helps. Each time I have to check something in the Messenger doc, having this link really helps to find the big article with everything explained.

**Comment by javiereguiluz** [comment]
Feedback is mostly negative, so let's close this as "won't fix".

**Comment by epitre** [comment]
Hi @javiereguiluz,

I see you closed the issue because of "mostly negative" feedback.

Hmmm, I would kindly disagree. 

2 people approved the PR @94noni and @alexandre-daubois .

And from my POV, I think it's better than before.


But maybe I missed a discussion than happened somewhere else.


Anyway, I just wanted to point out that the feedback here is not "mostly negative" IMHO.

Best Regards

**Pull request closed without merging**